### PR TITLE
themes

### PR DESCRIPTION
**Pull request by ltclmbdev**

### CODE REVIEW

File: src/components/ThemeToggler.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import React from 'react'
+
+const ThemeToggler = ({ toggleTheme, themeName }) => (
```
Comment (ltclmbdev) [inline src/components/ThemeToggler.js]: it's demo component, just to show how themes toggle

**Comment by ltclmbdev** [comment]
@joelhooks don't merge this branch, it's just for demo of how themes work.
If you are okay with it in general - Vojta and me will apply proper colors and so on.

File: src/components/Layout.js [hunk 2]
```diff
@@ -1,119 +1,25 @@
...
-      border-radius: 5px;
-    }
-  }
-  ${reset};
-`
+const Layout = ({ site, frontmatter = {}, children, noFooter }) => {
+  const [themeName, setTheme] = useState('default')
+  const toggleTheme = name => setTheme(name)
+
+  const theme = { ...themes[themeName], toggleTheme: toggleTheme }
```
Comment (ltclmbdev) [inline src/components/Layout.js]: Extend theme object with toggleTheme handler here. So we are able to switch the theme from almost any component.

File: src/components/Header.js [hunk 3]
```diff
@@ -1,99 +1,101 @@
 import React from 'react'
 import { Link } from 'gatsby'
 import { css } from '@emotion/core'
-import theme from '../../config/theme'
+import { useTheme } from '../../config/theming'
```
Comment (joelhooks) [inline src/components/Header.js]: I'd consider putting this in /components instead of config...

Cleaner import and since it will be used quite a bit...

**Review by joelhooks — APPROVED** [review]

File: src/components/Header.js (hunk 3 above)
Comment (ltclmbdev) [reply src/components/Header.js]: Gotcha, will fix.

**Pull request merged**